packages/react-storage/src/components/StorageBrowser/context/actions/__tests__/listLocationsAction.spec.ts

```diff
@@ -71,7 +71,72 @@ describe('createListLocationsAction', () => {
     expect(nextToken).toBeDefined();
   });
 
+  it('should paginate with default page limit and provide next token', async () => {
+    // assume, total items: 1500; default page limit: 1000
+    mockListLocations.mockResolvedValueOnce({
+      locations: generateMockLocations(600),
+      nextToken: 'next',
+    });
+    mockListLocations.mockResolvedValueOnce({
+      locations: generateMockLocations(400),
+      nextToken: 'next',
+    });
+
+    const listLocationsAction = createListLocationsAction(mockListLocations);
+
+    const output = await listLocationsAction(
+      { nextToken: undefined, result: [] },
+      {}
+    );
+
+    expect(mockListLocations).toHaveBeenNthCalledWith(1, {
+      pageSize: 1000,
+      nextToken: undefined,
+    });
+
+    expect(mockListLocations).toHaveBeenLastCalledWith({
+      pageSize: 400,
+      nextToken: 'next',
+    });
+
+    expect(mockListLocations).toHaveBeenCalledTimes(2);
+    expect(output.result).toHaveLength(1000);
+    expect(output.nextToken).toBeDefined();
+  });
+
+  it('should paginate with input page limit and conclude', async () => {
+    // assume, total items: 70; requested page limit: 100
+    mockListLocations.mockResolvedValueOnce({
+      locations: generateMockLocations(50),
+      nextToken: 'next',
+    });
+    mockListLocations.mockResolvedValueOnce({
+      locations: generateMockLocations(20),
+      nextToken: undefined,
+    });
+
+    const listLocationsAction = createListLocationsAction(mockListLocations);
+
+    const output = await listLocationsAction(
+      { nextToken: undefined, result: [] },
+      { options: { pageSize: 100 } }
+    );
+
+    expect(mockListLocations).toHaveBeenNthCalledWith(1, {
+      pageSize: 100,
+      nextToken: undefined,
+    });
+
+    expect(mockListLocations).toHaveBeenLastCalledWith({
+      pageSize: 50,
+      nextToken: 'next',
+    });
```

Comment:
Prefer to see seperate assertions, e.g. `toHaveBeenCalledTimes(expected_number)`, `toHaveBeenCalledWIth(expected_input)` but if we were to use `toHaveBeenNthCalledWith` it should be used for all assertions within a test block for consistency

Reply:
thanks makes sense, updated